cub/cub/device/dispatch/tuning/tuning_transform.cuh

```diff
@@ -207,130 +256,225 @@ _CCCL_HOST_DEVICE constexpr auto loaded_bytes_per_iteration() -> int
     smem_size += max_bytes_to_copy;
   };
   return smem_size;
 }
 
 constexpr int bulk_copy_size_multiple = 16;
 
-_CCCL_HOST_DEVICE constexpr auto bulk_copy_alignment(int sm_arch) -> int
+_CCCL_HOST_DEVICE constexpr auto bulk_copy_alignment(::cuda::arch_id arch) -> int
 {
-  return sm_arch < 1000 ? 128 : 16;
+  return arch < ::cuda::arch_id::sm_100 ? 128 : 16;
 }
 
-template <typename ItValueSizesAlignments>
-_CCCL_HOST_DEVICE constexpr auto
-bulk_copy_dyn_smem_for_tile_size(ItValueSizesAlignments it_value_sizes_alignments, int tile_size, int bulk_copy_align)
-  -> int
+template <int InputCount>
+_CCCL_HOST_DEVICE constexpr auto bulk_copy_dyn_smem_for_tile_size(
+  const ::cuda::std::array<iterator_info, InputCount>& inputs, int tile_size, int bulk_copy_align) -> int
 {
   // we rely on the tile_size being a multiple of alignments, so shifting offsets/pointers by it retains alignments
   _CCCL_ASSERT(tile_size % bulk_copy_align == 0, "");
   _CCCL_ASSERT(tile_size % bulk_copy_size_multiple == 0, "");
 
   int tile_padding = bulk_copy_align;
-  for (auto&& [_, vt_alignment] : it_value_sizes_alignments)
+  for (const auto& input : inputs)
   {
-    tile_padding = ::cuda::std::max(tile_padding, static_cast<int>(vt_alignment));
+    tile_padding = ::cuda::std::max(tile_padding, input.value_type_alignment);
   }
 
   int smem_size = tile_padding; // for the barrier and padding
-  for (auto&& [vt_size, _] : it_value_sizes_alignments)
+  for (const auto& input : inputs)
   {
-    smem_size += tile_padding + static_cast<int>(vt_size) * tile_size;
+    smem_size += tile_padding + input.value_type_size * tile_size;
   }
   return smem_size;
 }
 
-_CCCL_HOST_DEVICE constexpr int arch_to_min_bytes_in_flight(int sm_arch)
+_CCCL_HOST_DEVICE constexpr int arch_to_min_bytes_in_flight(::cuda::arch_id arch)
 {
-  if (sm_arch >= 1000)
+  if (arch >= ::cuda::arch_id::sm_100)
   {
     return 64 * 1024; // B200
   }
-  if (sm_arch >= 900)
+  if (arch >= ::cuda::arch_id::sm_90)
   {
     return 48 * 1024; // 32 for H100, 48 for H200
   }
-  if (sm_arch >= 800)
+  if (arch >= ::cuda::arch_id::sm_80)
   {
     return 16 * 1024; // A100
   }
   return 12 * 1024; // V100 and below
 }
 
-template <typename T>
-inline constexpr size_t size_of = sizeof(T);
-
-template <>
-inline constexpr size_t size_of<void> = 0;
-
-template <typename... RandomAccessIteratorsIn>
-_CCCL_HOST_DEVICE static constexpr auto make_sizes_alignments()
-{
-  return ::cuda::std::array<::cuda::std::pair<::cuda::std::size_t, ::cuda::std::size_t>,
-                            sizeof...(RandomAccessIteratorsIn)>{
-    {{sizeof(it_value_t<RandomAccessIteratorsIn>), alignof(it_value_t<RandomAccessIteratorsIn>)}...}};
-}
-
-template <int PtxVersion, int StoreSize, int... LoadSizes>
-struct tuning_vec
+[[nodiscard]] _CCCL_API constexpr auto vectorized_policy_for_filling(::cuda::arch_id arch, int store_size)
```

Comment:
This is using the blessed `_CCCL_API ` which is not used above. I believe we should be consistent